alws/utils/oval_add_gpg_keys.py

```diff
@@ -0,0 +1,289 @@
+      <red-def:object object_ref="oval:org.almalinux.alsa:obj:20235928001" />
+      <red-def:state state_ref="oval:org.almalinux.alba:ste:20191992002" />
+    </red-def:rpminfo_test>
+  </tests>
+  <objects>
+    <red-def:rpminfo_object id="oval:org.almalinux.alsa:obj:20235928001" version="636">
+      <red-def:name>tomcat</red-def:name>
+    </red-def:rpminfo_object>
+  <objects>
+  <states>
+    <red-def:rpminfo_state id="oval:org.almalinux.alba:ste:20191992002" version="635">
+      <red-def:signature_keyid operation="equals">51d6647ec21ad6ea</red-def:signature_keyid>
+    </red-def:rpminfo_state>
+  </states>
+
+To this:
+<oval_definitions>
+  <definition>
+    <criteria operator="AND">
+      <criterion test_ref="oval:org.almalinux.alsa:tst:20237065001"
+        comment="tomcat is earlier than 1:9.0.62-27.el8_9" />
+      <criteria operator="OR">
+        <criterion test_ref="oval:org.almalinux.alsa:tst:20235928002"
+          comment="tomcat is signed with AlmaLinux OS 8 key (51d6647ec21ad6ea)" />
+        <criterion test_ref="oval:org.almalinux.alsa:tst:20235928003"
+          comment="tomcat is signed with AlmaLinux OS 8 key (2ae81e8aced7258b)" />
+      </criteria>
+    </criteria>
+  <definitions>
+</oval_definitions>
+<tests>
+  <red-def:rpminfo_test check="at least one"
+    comment="tomcat is signed with AlmaLinux OS 8 key (51d6647ec21ad6ea)"
+    id="oval:org.almalinux.alsa:tst:20235928002" version="636">
+    <red-def:object object_ref="oval:org.almalinux.alsa:obj:20235928001" />
+    <red-def:state state_ref="oval:org.almalinux.alba:ste:20191992002" />
+  </red-def:rpminfo_test>
+  <red-def:rpminfo_test check="at least one"
+    comment="tomcat is signed with AlmaLinux OS 8 key (2ae81e8aced7258b)"
+    id="oval:org.almalinux.alsa:tst:20235928003" version="636">
+    <red-def:object object_ref="oval:org.almalinux.alsa:obj:20235928001" />
+    <red-def:state state_ref="oval:org.almalinux.alba:ste:20191992003" />
+  </red-def:rpminfo_test>
+</tests>
+<objects>
+<red-def:rpminfo_object id="oval:org.almalinux.alsa:obj:20235928001" version="636">
+  <red-def:name>tomcat</red-def:name>
+</red-def:rpminfo_object>
+<objects>
+<states>
+  <red-def:rpminfo_state id="oval:org.almalinux.alba:ste:20191992002" version="635">
+    <red-def:signature_keyid operation="equals">51d6647ec21ad6ea</red-def:signature_keyid>
+  </red-def:rpminfo_state>
+  <red-def:rpminfo_state id="oval:org.almalinux.alba:ste:20191992003" version="635">
+    <red-def:signature_keyid operation="equals">2ae81e8aced7258b</red-def:signature_keyid>
+  </red-def:rpminfo_state>
+</states>
+"""
+
+import itertools
+import re
+from copy import deepcopy
+from typing import Any, Dict, List, Optional, Tuple
+
+from almalinux.liboval.composer import Composer, Definition, RpminfoState, RpminfoTest
+
+# Map for GPG key ids for different distributions
+GPG = {8: ['51d6647ec21ad6ea', '2ae81e8aced7258b'], 9: ['d36cb86cb86b3716']}
```

Comment:
I'd say we no don't need such map nor passing the `distr_version` param around, as you indicated earlier, only AlmaLinux 8 is affected.

Reply:
@javihernandez 
 I added this as support for 9 distr (and all other future distr) in the future, so maybe we should keep it
What do you think?

Reply:
I know, and kind of makes sense, but the almalinux 8 key change was such an unfortunate event that the core team have put actions in place in order to avoid this situation again in the future. As I said, it's a minor comment aiming to simplify things a bit, I'm fine if you want to keep it, but personally I don't see this happening again in the future

Reply:
@javihernandez , ok)
I`ve  applied your recommendations